pkg/resources/kafka/configmap.go

```diff
@@ -346,6 +346,44 @@ func generateListenerSSLConfig(config *properties.Properties, name string, sslCl
 	}
 }
 
+// mergeSuperUsersPropertyValue merges the super.users property value from the source into the target properties
+func mergeSuperUsersPropertyValue(source *properties.Properties, target *properties.Properties) {
+	sourceVal, foundSource := source.Get("super.users")
+	if !foundSource || sourceVal.IsEmpty() {
+		return
+	}
+	targetVal, foundTarget := target.Get("super.users")
+	if !foundTarget || targetVal.IsEmpty() {
+		return
+	}
+
+	sourceSuperUsers := strings.Split(sourceVal.Value(), ";")
+	targetSuperUsers := strings.Split(targetVal.Value(), ";")
+
+	inserted := false
+	for _, sourceSuperUser := range sourceSuperUsers {
+		found := false
+		for _, targetSuperUser := range targetSuperUsers {
+			if sourceSuperUser == targetSuperUser {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			inserted = true
+			targetSuperUsers = append(targetSuperUsers, sourceSuperUser)
+		}
+	}
+
+	if inserted {
+		mergedSuperUsers := strings.Join(targetSuperUsers, ";")
+		// Setting string value for a property is not going to run into error, also we don't have error handling at upper levels
+		//nolint:errcheck
+		target.Set("super.users", mergedSuperUsers)
+	}
+}
+
```

Comment:
**Recommended**: I think the current implementation is a little more complex than it should be.
For a `len(source) == n` and `len(target) == m` and `intersect(source, target) == o` the complexity is somewhere between `O(n*m)` and `O(n*(m-o))`.
1. If the super.Users lists were sorted (or could be sorted and not taking into consideration the complexity of the sorting) then a simple merging algorithm would work faster here (iterating both slices with one cursor taking the earliest item), namely `O(n+m)`.
2. or If the super.Users could not be sorted easily then we could still use a map to merge the unique values into one comprehensive list which would make the complexity `O(n+m)` with an extra `union(n, m) < n+m` memory requirement.

Reply:
On paper you have right regarding performance and complexity, but in practically in our case my method is the fastest.
The hash function also has complexity and more memory allocated when u create a map data structure.
Sorting also has complexity as you said. 
In our case because of the small sizes of the source and target slices my method is the fastest and it has the lowest memory requirements.

BenchmarkMapMethod-8             2700618               447.6 ns/op           432 B/op          5 allocs/op
BenchmarkMyMethod-8             11876641                99.88 ns/op          128 B/op          1 allocs/op
BenchmarkSortMethod-8            2892297               412.7 ns/op           544 B/op          7 allocs/op

Other question which one is the most beautiful solution :)

Reply:
Try with the following mapMethod pls:

```golang
func mapMethod(a []string, b []string) (res []string) {
	check := make(map[string]interface{}, len(a)+len(b))
	res = make([]string, 0, len(a)+len(b))
	
	for _, item := range a {
		check[item] = nil
	}
	
	for _, item := range b {
		check[item] = nil
	}

	for letter, _ := range check {
		res = append(res, letter)
	}

	return res
}
```

If the result is similar then I let this go on the basis of being opinionated.

Reply:
When the input is:
a = []string{"alma", "korte", "User:CN=kafka-kafka-cluster", "User:CN=kafka-kafka-operator", "User:CN=supertubes-system-supertubes", "User:CN=supertubes-system-supertubes-ui"}
b = []string{"User:CN=kafka-kafka-cluster", "User:CN=kafka-kafka-operator", "User:CN=supertubes-system-supertubes", "User:CN=supertubes-system-supertubes-ui"}
result:
BenchmarkMapMethod-8            33648338               351.9 ns/op           432 B/op          5 allocs/op
BenchmarkMapMethod2-8           33422253               356.9 ns/op           736 B/op          2 allocs/op
BenchmarkMyMethod-8             148691840               80.95 ns/op          128 B/op          1 allocs/op
BenchmarkSortMethod-8           35608514               332.0 ns/op           544 B/op          7 allocs/op

When 
a =  []string{"alma", "korte"}
b = []string{"User:CN=kafka-kafka-cluster", "User:CN=kafka-kafka-operator", "User:CN=supertubes-system-supertubes", "User:CN=supertubes-system-supertubes-ui"}

BenchmarkMapMethod-8            38433975               315.2 ns/op           336 B/op          5 allocs/op
BenchmarkMapMethod2-8           70905468               167.6 ns/op            96 B/op          1 allocs/op
BenchmarkMyMethod-8             218245809               54.87 ns/op          128 B/op          1 allocs/op
BenchmarkSortMethod-8           51449380               234.7 ns/op           288 B/op          6 allocs/op

Reply:
Fine, I let this go.